### PR TITLE
Close #172 #158 #165 Doc update for PR template, description, and R Version in programming strategy

### LINKED ISSUE #172: Documentation: Update PR Checklist Template (Edge Unit Tests and keywords/families)
### Please select a category the issue is focused on?

Other

### Let us know where something needs a refresh or put your idea here!

Equivalent to https://github.com/pharmaverse/admiral/issues/1331 but for admiraldev repo

PR checklist should include a nudge towards developers looking at edge cases in their tests, e.g. empty datasets, errors, etc.

PR checklist should also include a link to the list of approved keywords and families allowed in an admiral function. Developers and reviewers should adhere to this list. If a new one is needed, then a proposal should be made.

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page.
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging.
- [x] Address all merge conflicts and resolve appropriately.
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,18 +1,19 @@
-Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.
+
```
Comment (bms63) [inline .github/pull_request_template.md:1]: new line?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
 
 - [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
 - [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
-- [ ] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html)
-- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
-- [ ] Update to all relevant roxygen headers and examples.
+- [ ] Updated relevant unit tests or have written new unit tests, which should include edge cases, e.g. empty datasets, errors, etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
```
Comment (bms63) [inline .github/pull_request_template.md:8]:
```suggestion
- [ ] Updated relevant unit tests or have written new unit tests, which should include edge cases, e.g. empty datasets, errors, etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
```

Comment (bms63) [reply .github/pull_request_template.md:8]: actually does this link work?

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
+- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation-1)?
```
Comment (bms63) [inline .github/pull_request_template.md:9]:
```suggestion
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation-1)?
```

Comment (bms63) [reply .github/pull_request_template.md:9]: can you make sure all these links work please

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
+- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
 - [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
 - [ ] Address any updates needed for vignettes and/or templates
 - [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
-- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/devel/reference/index.html)" page.
+- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page.
```
Comment (bms63) [inline .github/pull_request_template.md:14]: I guess since this admiraldev we should point to the Reference for that package down

File: DESCRIPTION [hunk 5]
```diff
@@ -11,6 +11,10 @@ Authors@R: c(
     person("Syed", "Mubasheer", role = "aut"),
     person("Ross", "Farrugia", role = "aut"),
     person("Ondrej", "Slama", role = "ctb"),
+    person("Saddchla", "Mascary", role = "ctb"),
```
Comment (sadchla-codes) [inline DESCRIPTION:14]: minor typo, first name is supposed to spelled `Sadchla` with 1 less `d`

File: DESCRIPTION [hunk 6]
```diff
@@ -11,6 +11,10 @@ Authors@R: c(
...
+    person("Zelos", "Zhu", role = "ctb"),
+    person("Jeffrey", "Dickinson", role = "ctb"),
+    person("Ania", "Golab", role = "ctb"),
```
Comment (bms63) [inline DESCRIPTION:17]: change to `aut`

**Review by sadchla-codes — CHANGES_REQUESTED** [review]

**Comment by bms63** [comment]
Also bump you guys up above the `ctb` folks.

File: .github/pull_request_template.md (hunk 1 above)
Comment (zdz2101) [reply .github/pull_request_template.md:1]: I had copy/pasted the exact same template from admiral, I realize now they were actually different?

File: .github/pull_request_template.md (hunk 3 above)
Comment (zdz2101) [reply .github/pull_request_template.md:9]: Links indeed work, small caveat is that /devel/ website can go back to /main/ website, but not the other way around, but I think this is a separate issue

Comment (bms63) [reply .github/pull_request_template.md:9]: yes i think that has to do with the `readme.Rmd` file.  I think there is an issue to investigate

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -1,18 +1,18 @@
...
 - [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
-- [ ] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html)
-- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
-- [ ] Update to all relevant roxygen headers and examples.
+- [ ] Updated relevant unit tests or have written new unit tests, which should include edge cases, e.g. empty datasets, errors, etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
+- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation-1)?
```
Comment (sadchla-codes) [inline .github/pull_request_template.md:8]: is the `-1` here needed ?`https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation-1`

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -1,18 +1,18 @@
...
 
 Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.
 
 - [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
 - [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
-- [ ] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html)
+- [ ] Updated relevant unit tests or have written new unit tests, which should include edge cases, e.g. empty datasets, errors, etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
```
Comment (sadchla-codes) [inline .github/pull_request_template.md:7]: The Unit Test guide doesn't mention any "edge cases", what exactly is an edge case? and should it be included in here?

Comment (zdz2101) [reply .github/pull_request_template.md:7]: Maybe I should use https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios ?

Comment (sadchla-codes) [reply .github/pull_request_template.md]: I think the entire document is relevant so I'm not sure I would point to a specific section of the unit test guide, but are there any documentations that provide further details about edge cases, if not I think pointing to the unit guide might be better..

@bms63 thoughts?

Comment (bms63) [reply .github/pull_request_template.md]: I don't think we want to get to in the weeds with edge cases in the PR Template, but we could expand on them in the guidance.   For example, edge case might be NAs, partial data, points almost at the <= or >= cut-offs, weird characters that can pop up in data we deal with.  Perhaps a few examples from the test files would suffice??

I didn't realize we had a section on realistic data scenarios.

Comment (bms63) [reply .github/pull_request_template.md]: I think we wanted to prompt people to think about edge cases in teh PR templates - as this has been some incidents where we could of caught bugs earilier if we had been more thoughtful in our edge cases.

Comment (zdz2101) [reply .github/pull_request_template.md]: Looks like I may have pulled the trigger a little too quick there for another commit. I also thought it was checkbox with the link was more as a reminder and once you comb over the unit-test doc a couple times it feels clear to try and come up with "weird" test cases

Comment (zdz2101) [reply .github/pull_request_template.md]: Should I just add an additional blurb on the guidance for this PR as well or make it a separate issue? @bms63

File: .github/pull_request_template.md [hunk 9]
```diff
@@ -1,18 +1,18 @@
...
 
 Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.
 
 - [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
 - [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
-- [ ] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html)
+- [ ] Updated relevant unit tests or have written new unit tests, which should include edge cases, e.g. empty datasets, errors, etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
```
Comment (bms63) [inline .github/pull_request_template.md:7]:
```suggestion
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
```

Comment (bms63) [reply .github/pull_request_template.md:7]: what about this?

Comment (zdz2101) [reply .github/pull_request_template.md:7]: Make sense to me to match previously written "lingo"

Comment (sadchla-codes) [reply .github/pull_request_template.md:7]: I think that sounds good!

**Review by sadchla-codes — APPROVED** [review]
